### PR TITLE
Add default tags to iam.Role

### PR DESCRIPTION
**Pull request by cebernardi**
### Description of your changes

This PR adds the default Crossplane tags ("crossplane-kind", "crossplane-name", "crossplane-providerconfig") to IAMRoles

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
- Unit-tests
- Manual testing:
    - spin up a new kind cluster (`kind create cluster --name crossplane-aws`)
    - install CRDs (`kubectl apply -f package/crds`)
    - *create resource scenario*:
        - create an iam.Role in the cluster (`kubectl apply -f examples/iam/role.yaml`)
        - start `crossplane-aws` controller
        - verify that the tags "crossplane-kind", "crossplane-name", "crossplane-providerconfig" are added as expected
    - *modify resource scenario*:
        - delete tags from the resource: `kubectl patch role.iam.aws.crossplane.io/somerole  --type=json -p='[{"op": "remove", "path":"/spec/forProvider/tags"}]'`
        - verify that the tags "crossplane-kind", "crossplane-name", "crossplane-providerconfig" are re-added as expected
   - *start controller scenario*:
       - stop `provider-aws` controller
       - reset the iam.Role (`kubectl apply -f examples/iam/role.yaml`) 
       - start `crossplane-aws` controller
       - verify that the tags "crossplane-kind", "crossplane-name", "crossplane-providerconfig" are added as expected

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by muvaf** [comment]
Thanks @cebernardi ! https://github.com/crossplane/provider-aws/pull/996 moves `IAMRole` in `identity` group to `iam` group with kind name `Role` and I expect that PR to land in the next few days. Can we hold this one and rebase once that's merged?

**Comment by cebernardi** [comment]
yes sure!

**Comment by muvaf** [comment]
@cebernardi It's merged, we should be good to rebase this now!

**Comment by cebernardi** [comment]
nice, I'll rebase ASAP

**Comment by cebernardi** [comment]
@muvaf I rebased and squashed

File: pkg/controller/iam/role/controller_test.go
Comment (muvaf) [inline pkg/controller/iam/role/controller_test.go:25]:
```suggestion
	"github.com/google/go-cmp/cmp/cmpopts"
	"github.com/aws/aws-sdk-go-v2/aws"
```

File: pkg/controller/iam/role/controller_test.go
Comment (muvaf) [inline pkg/controller/iam/role/controller_test.go:42]: nit: this should go to the third party imports block.

**Review by muvaf — APPROVED** [review]
Thank you @cebernardi and congrats on the 1000th PR!! It should be good to merge after fixing those small import issues.

**Comment by muvaf** [comment]
@cebernardi we'll need you to sign DCO since it's your first contribution https://github.com/crossplane/provider-aws/pull/1000/checks?check_run_id=4535934088

File: pkg/controller/iam/role/controller_test.go
Comment (cebernardi) [reply pkg/controller/iam/role/controller_test.go:41]: @muvaf I'm fixing it...
but I'm curious, shouldn't `make reviewable` take care of these issues?
I'm running fmt but it doesn't find anything to change, fmtimport as well..
silly question, but can you recommend a tool I can use?

File: pkg/controller/iam/role/controller_test.go
Comment (cebernardi) [reply pkg/controller/iam/role/controller_test.go]: actually, my local fmt reverts this change back... could you please give me the fmt configuration you're using?

File: pkg/controller/iam/role/controller_test.go
Comment (cebernardi) [reply pkg/controller/iam/role/controller_test.go]: please ignore me :)

**Comment by cebernardi** [comment]
I should have incorporated your suggestions, @muvaf

**Review by haarchri — APPROVED** [review]
LGTM - tested it - tagging is working - thanks ;)

**Pull request merged**
